ckanext/fairdatapoint/profiles.py

```diff
@@ -144,3 +146,35 @@ def _parse_contact_point(self, dataset_dict: Dict, dataset_ref: URIRef) -> Dict:
             dataset_dict['extras'] = \
                 [item for item in dataset_dict['extras'] if item.get('key') not in dcat_profile_contact_fields]
         return dataset_dict
+
+    def _parse_creator(self, dataset_dict: Dict, dataset_ref: URIRef) -> Dict:
+        graph = self.g
+        creators = []
+        for creator_ref in graph.objects(dataset_ref, DCTERMS.creator):
+            creator = {}
+            creator_identifier = graph.value(creator_ref, DCTERMS.identifier)
+            creator_name = graph.value(creator_ref, FOAF.name)
+
+            if creator_identifier:
+                creator['creator_identifier'] = str(creator_identifier)
+            if creator_name:
+                creator['creator_name'] = str(creator_name)
+            else:
+                # If the creator is a URI, use it as the identifier
+                if isinstance(creator_ref, URIRef):
+                    creator['creator_identifier'] = str(creator_ref)
+                    # Attempt to get the name from the graph if available
+                    name_from_graph = graph.value(creator_ref, FOAF.name)
+                    if name_from_graph:
+                        creator['creator_name'] = str(name_from_graph)
+                    else:
+                        creator['creator_name'] = str(creator_ref)
```

Comment:
`creator_name` is already available, is there added value for getting it here? `creator_ref` is the same here as above, so it will resolve to the same value.

Reply:
check